### PR TITLE
Upgrade TODO from pylint 2.x to pylint 3.0

### LINKED ISSUE #8335: There are three TODO's that are related to version 2.16 in pylint
### Current problem

As I was looking through the pylint code, I noticed that there were three TODO's that are about pylint version 2.16.

I thought that it would be good to add that information as an issue

The TODO's and locations are:

- "pylint/checkers/variables.py": # TODO: 2.16: Consider using RefactoringChecker._is_function_def_never_returning
- "pylint/reporters/json_reporter.py": TODO: 2.16: Add a new JSONReporter handling abs-path, confidence and scores. (Ultimately all other breaking change related to json for 3.0)
- "script/fix_documentation.py": # TODO: 2.16.0: Upgrade script for change in changelog

### Desired solution

1. The TODO's are fixed
2. The TODO's have an issue that discuss them that are added to a miletone for a pylint version that is 2.16.x
3. The TODO's are moved to be done in a later version

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Work in progress to discuss #8335 for 3.0.0a6

Closes #8335

### CODE REVIEW

File: pylint/reporters/json_reporter.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ class JSONReporter(BaseJSONReporter):
     TODO: 3.0: Remove this JSONReporter in favor of the new one handling abs-path
     and confidence.
 
-    TODO: 2.16: Add a new JSONReporter handling abs-path, confidence and scores.
+    TODO: 3.0: Add a new JSONReporter handling abs-path, confidence and scores.
```
Comment (Pierre-Sassoulas) [inline pylint/reporters/json_reporter.py:78]: Here I think we should just upgrade the TODO, we can't release a new 2.X version easily, let's just do it in 3.0.

**Review by DanielNoord — APPROVED** [review]
Let's just move them for now.

File: pylint/testutils/functional/test_file.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ class FunctionalTestFile:
     def __init__(self, directory: str, filename: str) -> None:
         self._directory = directory
         self.base = filename.replace(".py", "")
-        # TODO: 2.x: Deprecate FunctionalTestFile.options and related code
+        # TODO: 3.0: Deprecate FunctionalTestFile.options and related code
         # We should just parse these options like a normal configuration file.
```
Comment (Pierre-Sassoulas) [inline pylint/testutils/functional/test_file.py:66]: What did you mean by that @DanielNoord ? Just removing self.options is not possible there's the line in `` self.options[name] = conv(value)  # type: ignore[literal-required]`` not sure what to do exactly.

File: pylint/checkers/variables.py [hunk 3]
```diff
@@ -2461,7 +2461,7 @@ def _loopvar_name(self, node: astroid.Name) -> None:
                 else_stmt, (nodes.Return, nodes.Raise, nodes.Break, nodes.Continue)
             ):
                 return
-            # TODO: 2.16: Consider using RefactoringChecker._is_function_def_never_returning
+            # TODO: 3.0: Consider using RefactoringChecker._is_function_def_never_returning
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/variables.py:2464]: No idea what to do at all here, it seems like a major refactor. We can do it, push it to another milestone or not do it (?)

File: pylint/testutils/functional/test_file.py (hunk 2 above)
Comment (DanielNoord) [reply pylint/testutils/functional/test_file.py:66]: We shouldn't need to rely on `options` but rather just have the test run `pylint test_file.py --rcfile=test_file.rc` We have similar logic for the doc tests.

Comment (Pierre-Sassoulas) [reply pylint/testutils/functional/test_file.py:66]: Ha right, thank you.

**Comment by Pierre-Sassoulas** [comment]
Took me way too long to realize that there was a ``[ci skip]`` in the commit message because I did not expect to merge as is.

**Review by DanielNoord — APPROVED** [review]
Didn't even know we could skip CI

**Pull request merged**